File: Java/ToyLanguageGUI_exam_final/src/domain/expressions/VarExpression.java
package domain.expressions;


import domain.expressions.Expression;
import utils.IDictionary;
import utils.IHeap;

public class VarExpression implements Expression {
    private String varName;
    public VarExpression(String varName){
        this.varName = varName;
    }

    @Override
    public int eval(IDictionary<String,Integer> symbTable, IHeap<Integer,Integer> heap) throws RuntimeException{
        if(symbTable.contains(varName))
            return symbTable.get(varName);
        throw new RuntimeException("Not found");
    }

    @Override
    public String toString(){
        return varName;
    }

}

File: ToyInterpretor/ToyInterpretor/Domain/Expressions/VarExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyInterpretor.Utils;

namespace ToyInterpretor.Domain.Expressions
{
    class VarExpression : Expression
    {
        private String varName;
        public VarExpression(String varName)
        {
            this.varName = varName;
        }

        public int eval(ISymTable<String, int> symbTable) // throws RuntimeException
        {
           if (symbTable.contains(varName))
              return symbTable.get(varName);
           throw new SystemException("Not found");
        }

            
        public override String ToString()
        {
            return varName;
        }
    }
}
